helper/ipsec.c

```diff
@@ -146,9 +197,11 @@ int odph_ipsec_alg_check(odp_ipsec_capability_t capa,
 		return -1;
 	}
 
+	default_icv_len = odph_ipsec_auth_icv_len_default(auth_alg);
 	found = false;
 	for (i = 0; i < num; i++) {
-		if (auth_capa[i].key_len == auth_key_len) {
+		if (auth_capa[i].key_len == auth_key_len &&
+		    auth_capa[i].icv_len == default_icv_len) {
```

Comment:
There's debug print a few lines below, saying "Unsupported auth key length", which could be misleading after this change.

Reply:
Improved (hopefully) the debug print in v2.